actions/list.action.ts

```diff
@@ -1,39 +1,52 @@
 import * as chalk from 'chalk';
 import * as fs from 'fs';
-import { platform, release } from 'os';
-import osName = require('os-name');
 import { join } from 'path';
-import {
-  AbstractPackageManager,
-  PackageManagerFactory,
-} from '../lib/package-managers';
-import { BANNER, MESSAGES } from '../lib/ui';
+import { AbstractPackageManager, PackageManagerFactory } from '../lib/package-managers';
 import { AbstractAction } from './abstract.action';
-
+import { MESSAGES } from '../lib/ui';
 
 export class ListAction extends AbstractAction {
   private manager!: AbstractPackageManager;
+  private specFilePath: string;
+
+  constructor() {
+    super();
+    this.specFilePath = 'spec.yaml'; // Default spec file path if not found in nest-cli.json
+  }
 
   public async handle() {
     this.manager = await PackageManagerFactory.find();
+    await this.loadSpecFilePath();
     await this.loadRegistry();
   }
-  private async loadRegistry(): Promise<void>  {
-        const stencilFilePath = join(process.cwd(), '.stencil');
-        try {
-            const stencilFileContent = await fs.promises.readFile(stencilFilePath, 'utf-8');
 
-            if (stencilFileContent.trim() === '') {
-                console.log(chalk.yellow('No modules found'));
-              } else {
-                console.log(chalk.green('Installed Modules:'));
-                console.log(stencilFileContent);
-              }
-        }catch (error) {
-            console.error(chalk.red('Error reading .stencil file'));
-        }
-        
+  private async loadSpecFilePath(): Promise<void> {
+    const nestCliPath = join(process.cwd(), 'nest-cli.json');
+    try {
+      const nestCliContent = await fs.promises.readFile(nestCliPath, 'utf-8');
+      const nestCliConfig = JSON.parse(nestCliContent);
+
+      if (nestCliConfig.specFile) {
+        this.specFilePath = nestCliConfig.specFile;
+      }
+    } catch (error) {
+      console.error(chalk.yellow(MESSAGES.NEST_CLI_NOT_FOUND));
     }
+  }
 
+  private async loadRegistry(): Promise<void> {
+    const stencilFilePath = join(process.cwd(), this.specFilePath);
+    try {
+      const stencilFileContent = await fs.promises.readFile(stencilFilePath, 'utf-8');
 
+      if (stencilFileContent.trim() === '') {
+        console.info(chalk.yellow(MESSAGES.NO_SPEC_FOUND));
```

Comment:
Since we are not erroring out what is the behaviour incase the spec file is empty? Can't seem to find it in the docs.

Reply:
Right!
Will update that test case and add the condition under stencil spec command

Reply:
if spec.yaml is empty I was doing console.info "No specification file found". Should I change it to "Spec file empty" ?